### DISCUSSION TITLE
Additional sections to CIP template

### DISCUSSION THREAD

**Discussion by bloated-devish** [deso-protocol/dips#33] [Medium CIPs]
Can we add three more sections to the template?

Alternate designs - Gives the community idea on why some design choices were skipped.
Data storage change - As more fields are added, we need to start ensuring these new fields don't increase gas fees so much (as well as making nodes disk heavy).
Acknowledgements - People often have others who contributed, but not authors. This is a better place to ensure proper crediting?

For a preview on how the CIP will look like, here is the preview. These 3 sections will be added at the end below Security Considerations.

---
cip: <to be assigned>
title: <CIP title>
author: <a list of the author's or authors' name(s) and/or username(s), or name(s) and email(s), e.g. (use with the parentheses or triangular brackets): FirstName LastName (@GitHubUsername), FirstName LastName <foo@bar.com>, FirstName (@GitHubUsername) and GitHubUsername (@GitHubUsername)>
discussions-url: <URL>
created: <date created on, in ISO 8601 (yyyy-mm-dd) format>
---

This is the suggested template for new CIPs. Note that a CIP number will be assigned by a core dev.

## One-Line Description

A very simple description that someone who's quickly scanning CIPs can use to
immediately understand what this change is doing.

## Explanation and Motivation
Provide a simple explanation of the CIP.

Include an explanation of the "why" behind this CIP. What problem does it solve?

## Specification
The technical specification should describe the new feature in deep technical detail. It should be detailed enough to allow for complete implementation.
  
Include the rationale behind any decisions. If there were any approaches that didn't work, feel free to explain them to help future devs understand why things are the way they are.

## Backwards Compatibility
All CIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The CIP must explain how the author proposes to deal with these incompatibilities.

## Test Cases
If this change requires more than just unit tests, include an explanation of how you're going to test your change to make sure it works.

## Security Considerations
All CIPs must contain a section that discusses the security
implications/considerations relevant to the proposed change. Include
possible attack vectors or ways this change could cause problems
down the road.

## Alternate designs considered
Provide a list of other ideas that you considered and their pros and cons.
This will help community understand the constraints under which other ideas were
not chosen.

## Data Storage Change
Identify the additional storage requirement in the blockchain due to this change.

## Acknowledgements
If there were additional community members that helped you guide / hone your
thoughts, or members whose feedback was constructive to make this CIP better.

**Comment by bloated-devish** [comment]
@maebeam This still needs a CIP since we are changing the CIP process, but would you be okay if we add the last 3 sections?

1) Alternate designs considered
2) Data storage change
3) Acknowledgements

If you are curious to see how the template change PR would look like, here it is:
https://github.com/bloated-devish/cips/blob/cc0aebc0e45ba0cc9908a498dfbb4f17e8055936/cip-template.md

Thanks\
Bloated

**Comment by tijno** [comment]
I am happy with these additions to the template - makes sense.

**Comment by maebeam** [comment]
I'm happy with these additions. Data Storage Change feels like it should be covered in Specification though and doesn't need its own section

**Reply by bloated-devish** [reply]
I specifically added Data Storage - since we are growing data a lot - and its good for CIP creators and core devs to get an opportunity to think if this will bloat the blockchain. It's much harder to go and fix mistakes that caused duplicate data or large data increase that overall increases cost of every post (usually which is unintended).

I'll go ahead and leave data storage out, but if you agree, I can follow up and add it as a sub section to specification (I agree it feels better to put it there), or just leave it out completely (although you'll hear bloated grumbling a bit lol).

**Reply by bloated-devish** [reply]
Thank you for approving my PR. We can close this medium CIP.
